### PR TITLE
Add time helper to FieldDef

### PR DESCRIPTION
**Pull request by christophebeling**
Added a litte helper (analogous to http://www.rubydoc.info/gems/factory_girl/file/GETTING_STARTED.md#Lazy_Attributes) to create times easily

### CODE REVIEW

**Comment by till** [comment]
@christophebeling do you want to re-base the PR against master?

**Comment by christophebeling** [comment]
Actually it was rebased, but nobody checked it for ~ 4 month. however, i rebased it on the current master

**Comment by till** [comment]
@christophebeling care to check out the failing test?

**Comment by christophebeling** [comment]
fixed tests

File: src/FactoryGirl/Provider/Doctrine/DateIntervalHelper.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+namespace FactoryGirl\Provider\Doctrine;
+
+
+class DateIntervalHelper
+{
+
+    const DATE_TIME = 1;
+    const TIMESTAMP = 2;
+    const STRING = 3;
```
Comment (till) [inline src/FactoryGirl/Provider/Doctrine/DateIntervalHelper.php]: The naming of `STRING` is probably a bad idea.

File: README.markdown [hunk 2]
```diff
@@ -1,7 +1,7 @@
 Factory Girl in PHP
 ===================
 
-[![Build Status](https://secure.travis-ci.org/breerly/factory-girl-php.png?branch=master)](http://travis-ci.org/breerly/factory-girl-php)
```
Comment (till) [inline README.markdown]: ;)

**Comment by till** [comment]
phpstorm docblocks could go as well.

**Comment by christophebeling** [comment]
Dock Blocks has been added

File: src/FactoryGirl/Provider/Doctrine/DateIntervalHelper.php [hunk 3]
```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: cebeling
+ * Date: 29/05/15
+ * Time: 11:35
+ */
```
Comment (glaubinix) [inline src/FactoryGirl/Provider/Doctrine/DateIntervalHelper.php]: can you please remove the phpstorm docblock here?

File: src/FactoryGirl/Provider/Doctrine/DateIntervalHelper.php [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+     * @param bool|false $negative
+     */
+    public function __construct(\DateTime $time, $negative = false)
+    {
+        $this->time = $time;
+        $this->negative = $negative;
+    }
+
+    /**
+     * @param $years
```
Comment (glaubinix) [inline src/FactoryGirl/Provider/Doctrine/DateIntervalHelper.php]: this should be ````@param int $years````

File: src/FactoryGirl/Provider/Doctrine/DateIntervalHelper.php [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            throw new \RuntimeException();
+        }
+        $interval = new \DateInterval('P'.$years.'Y');
+        $interval->invert = (int) $this->negative;
+        $this->time->add($interval);
+        return $this;
+    }
+
+    /**
+     * @param $months
```
Comment (glaubinix) [inline src/FactoryGirl/Provider/Doctrine/DateIntervalHelper.php]: this should be ````@param int $months````

File: src/FactoryGirl/Provider/Doctrine/DateIntervalHelper.php [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        }
+
+        $interval = new \DateInterval('P'.$months.'M');
+        $interval->invert = (int) $this->negative;
+        $this->time->add($interval);
+        return $this;
+    }
+
+    /**
+     * @param $days
```
Comment (glaubinix) [inline src/FactoryGirl/Provider/Doctrine/DateIntervalHelper.php]: this should be ````@param int @days````

File: tests/FactoryGirl/Tests/Provider/Doctrine/Fixtures/TimeTest.php [hunk 7]
```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: cebeling
+ * Date: 29/05/15
+ * Time: 17:33
+ */
```
Comment (glaubinix) [inline tests/FactoryGirl/Tests/Provider/Doctrine/Fixtures/TimeTest.php]: can you please remove the phpstorm docblock here?

File: tests/FactoryGirl/Tests/Provider/Doctrine/Fixtures/TimeTest.php [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+/**
+ * Class TimeTest
+ * @package FactoryGirl\Tests\Provider\Doctrine\Fixtures
+ */
+class TimeTest extends TestCase
+{
+    /**
+     *  
+     */
+    public function testGetTimePast()
```
Comment (glaubinix) [inline tests/FactoryGirl/Tests/Provider/Doctrine/Fixtures/TimeTest.php]: can you remove empty doc block please?

File: tests/FactoryGirl/Tests/Provider/Doctrine/Fixtures/TimeTest.php [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+            FieldDef::past()->days(2)->months(1)->years(3)->get(DateIntervalHelper::DATE_STRING),
+            'Error getting string'
+        );
+
+    }
+
+    /**
+     *
+     */
+    public function testGetTimeFuture()
```
Comment (glaubinix) [inline tests/FactoryGirl/Tests/Provider/Doctrine/Fixtures/TimeTest.php]: can you remove empty doc block please?

**Comment by christophebeling** [comment]
cleaned up docblocks

**Comment by till** [comment]
@christophebeling now do you mind squashing it into one to clean-up the history? :) LMK if you need help. I think that is all then! Thank you.

**Comment by christophebeling** [comment]
@till squashed to one commit and rebased on current master

**Comment by till** [comment]
Nice! :+1: And thanks a lot.

@glaubinix you are good to merge?

**Comment by glaubinix** [comment]
yes, looks good.

**Pull request merged**